lightpath/ui/widgets.py

```diff
@@ -177,21 +180,52 @@ def add_signal(self, signal):
         self.command_layout.insertWidget(count - 1, label, 0, Qt.AlignHCenter)
         self.command_layout.insertWidget(count, widget, 0, Qt.AlignHCenter)
 
+    def get_state_color(self) -> QColor:
+        """
+        Determine the icon color given the device state and path status
+        If device is an impediment: state_color['blocking']
+        If device is removed: state_color['removed']
+        If device is in, not blocking (mirrors): state_color['half_removed']
+        If device is unknown / errored: state_color['error']
+
+        The color of the labels should quickly point users to blocking devices,
+        while providing useful information about each device's state.
+
+        Could take a color map in the future?  Colorblind support?
+
+        Returns
+        -------
+        QColor
+            the color to apply to the icon
+        """
+        device_state, _ = find_device_state(self.device)
+        blocking_devices = self.path.blocking_devices
+
+        if device_state is DeviceState.Disconnected:
+            return state_colors['disconnected']
+        elif device_state is DeviceState.Error:
```

Comment:
Minor style suggestion - big blocks containing multiple `elif` aren't really necessary when there's `return` in each of the clauses:
```suggestion
        if device_state is DeviceState.Error:
```
Then `else` even becomes unnecessary